redis_cache/__init__.py

```diff
@@ -208,6 +209,43 @@ def __call__(self, fn):
         self.keys_key = f'{self.get_full_prefix()}:keys'
         self.original_fn = fn
 
+        if asyncio.iscoroutinefunction(fn):
+
+            @wraps(fn)
+            async def inner_async(*args, **kwargs):
```

Comment:
Why name this differently versus just using `inner`, so we can reuse these lines?

```python
inner.invalidate = self.invalidate
inner.invalidate_all = self.invalidate_all
inner.get_full_prefix = self.get_full_prefix
inner.instance = self
```